### ISSUE TITLE
Confused about `range_did_not_narrow`

### ISSUE THREAD

**Issue by ericphanson** [JuliaRegistries/RegistryCI.jl#180]
```julia
julia> ranges = [Pkg.Types.VersionRange("1.1 - 1.1.5"), Pkg.Types.VersionRange("1.1.7 - 1.2")]
2-element Array{Pkg.Types.VersionRange,1}:
 VersionRange("1.1-1.1.5")
 VersionRange("1.1.7-1.2")

julia> AutoMerge.range_did_not_narrow(ranges, ranges)
false
```

Is this right? I went to add some tests for this function but realized I didn't quite understand how this function is supposed to be used. I think this case (the two arguments are equal) is taken care of in the line above where `range_did_not_narrow` is called, but

```julia
julia> ranges2 = [Pkg.Types.VersionRange("1.1 - 1.1.5"), Pkg.Types.VersionRange("1.1.7 - 1.2"), Pkg.Types.VersionRange("1.2.1 - 1.3")]
3-element Array{Pkg.Types.VersionRange,1}:
 VersionRange("1.1-1.1.5")
 VersionRange("1.1.7-1.2")
 VersionRange("1.2.1-1.3")

julia> AutoMerge.range_did_not_narrow(ranges, ranges2)
false

julia> AutoMerge.range_did_not_narrow(ranges2, ranges)
false
```
also seems strange.

**Comment by DilumAluthge** [comment]
Yeah this function is terrible. Probably should be thrown away and rewritten from scratch.

The basic idea is that in a patch release, you shouldn't narrow your support for Julia, but you are allowed to widen your support for Julia.

**Comment by tkf** [comment]
I got this message

> A patch release is not allowed to narrow the supported ranges of Julia versions. The ranges have changed from Pkg.Types.VersionRange[VersionRange("0.7.0"), VersionRange("1")] (in 0.3.1) to Pkg.Types.VersionRange[VersionRange("0.7"), VersionRange("1")] (in 0.3.2).

in https://github.com/JuliaRegistries/General/pull/8014#issuecomment-574969632

Is it related to this issue?  It seems that AutoMerge detects the change from `VersionRange("0.7.0")` to `VersionRange("0.7")` as "narrowing" but IIUC it's actually "widening":

```julia
julia> in.([v"0.6.9", v"0.7.0", v"0.7.1"], Ref(Pkg.Types.VersionRange("0.7.0")))
3-element BitArray{1}:
 0
 1
 0

julia> in.([v"0.6.9", v"0.7.0", v"0.7.1"], Ref(Pkg.Types.VersionRange("0.7")))
3-element BitArray{1}:
 0
 1
 1
```

**Comment by DilumAluthge** [comment]
Yeah the `range_did_not_narrow` function is buggy and probably needs to be rewritten from scratch.
